source/SpinalHDL/Data types/bits.rst

```diff
@@ -242,52 +244,112 @@ To cast a ``Bool``, ``UInt`` or an ``SInt`` into a ``Bits``, you can use ``B(som
+All of the bit extraction operations can be used to read a bit / group of bits. Like in other HDLs They
+can also be used to select a range of bits to be written.
 
 .. list-table::
    :header-rows: 1
-   :widths: 2 5 2
+   :widths: 2 4 2
 
    * - Operator
      - Description
      - Return
-   * - x(y)
-     - Readbit, y: Int/UInt
+   * - x(y: Int)
+     - Static bit access of y-th bit
      - Bool
-   * - x(offset,width bits)
-     - Read bitfield, offset: UInt, width: Int
-     - Bits(width bits)
-   * - x(\ :ref:`range <range>`\ )
-     - Read a range of bit. Ex : myBits(4 downto 2)
-     - Bits(range bits)
-   * - x(y) := z
-     - Assign a single bit, y: Int/UInt
+   * - x(x: UInt)
+     - Variable bit access of y-th bit
      - Bool
-   * - x(offset, width bits) := z
-     - Assign bitfield, offset: UInt, width: Int
+   * - x(offset: Int, width bits)
+     - Fixed part select of fixed width, offset is LSB index
      - Bits(width bits)
-   * - x(\ :ref:`range <range>`\ ) := z
-     - Assign a range of bits. Ex : myBits(4 downto 2) := B"010"
-     - Bits(range bits)
+   * - x(offset: UInt, width bits)
+     - Variable part-select of fixed width, offset is LSB index
+     - Bits(width bits)
+   * - x(range: Range)
+     - Access a :ref:`range <range>` of bits. Ex : myBits(4 downto 2)
+     - Bits(range.size bits)
+   * - x.subdivideIn(y slices, [strict: Boolean])
+     - Subdivide x into y slices, y: Int
+     - Vec(Bits(...), y)
+   * - x.subdivideIn(y bits, [strict: Boolean])
+     - Subdivide x in multiple slices of y bits, y: Int
+     - Vec(Bits(y bit), ...)
+   * - x.msb
+     - Access most significant bit of x (highest index)
+     - Bool
+   * - x.lsb
+     - Access lowest significant bit of x (index 0)
+     - Bool
 
 
+Some basic examples:
+
 .. code-block:: scala
 
    // get the element at the index 4
    val myBool = myBits(4)
-
-   // assign
+   // assign element 1
    myBits(1) := True
 
-   // Range
-   val myBits_8bits = myBits_16bits(7 downto 0)
-   val myBits_7bits = myBits_16bits(0 to 6)
-   val myBits_6bits = myBits_16Bits(0 until 6)
+   // index dynamically
+   val index = UInt(2 bit)
+   val indexed = myBits(index, 2 bit)
+
+   // range index
+   val myBits_8bit = myBits_16bit(7 downto 0)
+   val myBits_7bit = myBits_16bit(0 to 6)
+   val myBits_6bit = myBits_16bit(0 until 6)
+   // assign to myBits_16bit(3 downto 0)
+   myBits_8bit(3 downto 0) := myBits_4bit
+
+   // equivalent slices, no reversing occurs
+   val a = myBits_16bit(8 downto 4)
+   val b = myBits_16bit(4 to 8)
+
+   // read / assign the msb / leftmost bit / x.high bit
+   val isNegative = myBits_16bit.msb
+   myBits_16bit.msb := False
+
+Subdivide details
+"""""""""""""""""
+
+Both overloads of ``subdivideIn`` have an optional parameter ``strict`` (i.e. ``subdivideIn(slices: SlicesCount, strict: Boolean = true)``).
+If ``strict`` is ``true`` an error will be raised if the input could not be divided evenly. If set to ``false`` the generated pieces may
```

Comment:
DUPLICATE COMMENT AS PER Int.rst FILE:444

`evenly` maybe `exactly` ?   use of even and odd terms start to become ambiguous when the slices!=2

`... the generated pieces may vary in size as necessary.`
`... the generated pieces may vary the size of the last slice as necessary.`

Reply:
as above